python-sdk/src/astro/sql/operators/load_file.py

```diff
@@ -95,6 +96,11 @@ def execute(self, context: Context) -> BaseTable | File:  # skipcq: PYL-W0613
         if self.output_table:
             context["ti"].xcom_push(key="output_table_conn_id", value=str(self.output_table.conn_id))
             context["ti"].xcom_push(key="output_table_name", value=str(self.output_table.name))
+        if self.skip_core_execution:
```

Comment:
@pankajkoti @tatiana I think we should not have this in the operators, there should be a better way of retrying a task.

Reply:
Also, this change will be required for all operators, I presume.

Reply:
@utkarsharma2 this was not introduced for retries..! Its goal is to allow us to execute a task by only populating the XComArg (e.g. it does not load, it just populate the output of execute with the output table, so we can resolve references in other tasks).

Reply:
This change would probably not make sense in the `run_raw_sql` - and we wanted to collect feedback on the overall approach before changing any other operators :)

Reply:
I am wondering if we are creating a workaround, instead of reading the information (xcom) from the db. I think the problem we have in our local runner is that we are not storing and instead actively deleting the previous dag runs in `_get_or_create_dagrun`. If we would keep them, we would be able to retrieve the previous xcom without re-running it (I think). @dimberman would that be an option? I think that would be a much cleaner solution if we can handle potential collisions which you mentioned in the function's docstring.

Reply:
We currently cannot access the previous xcoms, because the deletion of dagruns causes a _cascade delete_, deleting xcoms created by the task instances of the dagrun.

Reply:
@tatiana I see, assuming we have task `a -> b -> c` and now we want to rerun task `c`. We want to skip the `a` and `b` and just run `c` this is the use-case right?

Reply:
Exactly, @utkarsharma2 !

@feluelle I think it is not necessarily a workaround, I think there are a couple ways of addressing the problem - but let's try to see some pros and cons of each approach.

**1. Table name approach**

It doesn't rely on metadata from previous runs, but on the dynamic generation of table names.

Pros:
* Works even if the user didn't use the SQL CLI to populate the tables the task relies on
* Works across different users computers
* Could work even if the topology of the DAG changed

Cons:
* We had to change Python SDK operators -  this would not work for non-Python SDK operators
* This does not work for temporary tables

**2. Previous DAG Run approach**

Pros:
* We don't have to change the Python SDK operators - so it would be agnostic to the operator we use
* Works with temporary tables

Cons:
* Doesn't work if the table was populated outside of the SQL CLI
* Doesn't work cross-computers
* Would likely fail if the topology of the DAG changes

Would love to hear more pros and cons of each approach!

Reply:
One Con of the `Table name approach` is that we may be able to handle this for SDK operators this way(because we control code), but it won't work for Airflow's operator if we decide to use them later on in CLI.

Reply:
As suggested by @tatiana and @feluelle, we're now overriding the Python SDK operators in SQL CLI. So, there is no change in Python SDK operators now.